cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -0,0 +1,391 @@
+    getSignableRepresentation =
+      hashToBytes . hashWith @(HASH StandardCrypto) (serialiseToCBOR . asTxMetadata)
+
+instance AsTxMetadata GovernancePollAnswer where
+    asTxMetadata GovernancePollAnswer{govAnsPoll, govAnsChoice} =
+      makeTransactionMetadata $ Map.fromList
+        [ ( metadataLabel
+          , TxMetaMap $
+           [ ( TxMetaNumber 2, TxMetaBytes (serialiseToRawBytes govAnsPoll) )
+           , ( TxMetaNumber 3, TxMetaNumber (toInteger govAnsChoice) )
+           ]
+          )
+        ]
+
+instance SerialiseAsCBOR GovernancePollAnswer where
+    serialiseToCBOR =
+      serialiseToCBOR . asTxMetadata
+
+    deserialiseFromCBOR AsGovernancePollAnswer bs = do
+      metadata <- deserialiseFromCBOR AsTxMetadata bs
+      withNestedMap lbl metadataLabel metadata $ \values ->
+        GovernancePollAnswer
+          <$> case lookup (TxMetaNumber 2) values of
+                Nothing -> Left $ missingField (field "2")
+                Just x  -> expectHash (field "2") x
+          <*> case lookup (TxMetaNumber 3) values of
+                Just (TxMetaNumber n) -> expectWord (field "3") n
+                Just _  -> Left $ malformedField (field "3") "Number (uint)"
+                Nothing -> Left $ missingField (field "3")
+     where
+       lbl = "GovernancePollAnswer"
+       field i = lbl <> "." <> i
+
+       expectHash fieldLbl value =
+         case value of
+           TxMetaBytes bytes ->
+             left
+               (DecoderErrorCustom fieldLbl . Text.pack . unSerialiseAsRawBytesError)
+               (deserialiseFromRawBytes (AsHash AsGovernancePoll) bytes)
+           _ ->
+             Left (malformedField fieldLbl "Bytes")
+
+
+-- ----------------------------------------------------------------------------
+-- Governance Poll Witness
+--
+
+data GovernancePollWitness
+    = GovernancePollWitnessVRF
+        (VerKeyVRF StandardCrypto)
+        (VRF.CertVRF (VRF StandardCrypto))
+    | GovernancePollWitnessColdKey
+        (VKey 'Witness StandardCrypto)
+        (SignedDSIGN StandardCrypto GovernancePollAnswer)
+  deriving Show
+
+instance HasTypeProxy GovernancePollWitness where
+    data AsType GovernancePollWitness = AsGovernancePollWitness
+    proxyToAsType _ = AsGovernancePollWitness
+
+instance AsTxMetadata GovernancePollWitness where
+    asTxMetadata witness =
+      makeTransactionMetadata $ Map.fromList
+        [ ( metadataLabel
+          , TxMetaMap
+           [ case witness of
+              GovernancePollWitnessVRF vk proof ->
+                ( TxMetaNumber 4
+                , TxMetaList
+                    [ TxMetaBytes $ VRF.rawSerialiseVerKeyVRF vk
+                    , let bytes = VRF.rawSerialiseCertVRF proof in
+                      TxMetaList $
+                        [ TxMetaBytes $ BS.take txMetadataByteStringMaxLength bytes
+                        , TxMetaBytes $ BS.drop txMetadataByteStringMaxLength bytes
+                        ]
```

Comment:
I take it `bytes` never exceeds length `txMetadataByteStringMaxLength * 2`.

Reply:
Yes; Ed25519 signatures are 64 bytes and VRF proofs are 80 bytes.